twitchio/websocket.py

```diff
@@ -75,6 +75,15 @@ def __init__(self, **kwargs):
         self.nick = kwargs.get('nick').lower()
         self.modes = kwargs.pop('modes', ("commands", "tags", "membership"))
         self._initial_channels = kwargs.get('initial_channels')
+        
+        if callable(self._initial_channels):
+            _temp_initial_channels = self._initial_channels()
+            # If the result of the function is a list or tuple, assign it
+            # Otherwise turn it into a list and assign it
+            if type(_temp_initial_channels) == list or type(_temp_initial_channels) == tuple:
```

Comment:
Type checking should be done with isinstance as it's cleaner and doesn't fail on inherited types.

```suggestion
            if isinstance(_temp_initial_channels, (list, tuple)):
```

Reply:
Resolved with latest commit on branch, didn't realize this would let me auto commit requested changes 😞